RecentActivity/src/org/sleuthkit/autopsy/recentactivity/BinaryCookieReader.java

```diff
@@ -0,0 +1,429 @@
+                dataStream.skipBytes((2 * SIZEOF_INT_BYTES) + (pageSizeArray.length * SIZEOF_INT_BYTES));
+            } catch (IOException ex) {
+                logger.log(Level.WARNING, "Error occured creating DataInputStream", new Object[]{cookieFile.getName(), ex}); //NON-NLS
+                closeStream(); // Just incase the error was from skip
+            }
+        }
+
+        /**
+         * hasNext manages reading of each cookie page and creating the Cookie
+         * Page objects. If the currentIterator returns false from hadNext, read
+         * the next page and create and new instance of
+         * CookiePage\CookieIterator
+         *
+         * @return True if there are more cookies
+         */
+        @Override
+        public boolean hasNext() {
+
+            if (dataStream == null) {
+                return false;
+            }
+
+            if (currentIterator == null || !currentIterator.hasNext()) {
+                try {
+
+                    if (pageIndex < pageSizeArray.length) {
+                        byte[] nextPage = new byte[pageSizeArray[pageIndex]];
+                        dataStream.read(nextPage);
+
+                        currentPage = new CookiePage(nextPage);
+                        currentIterator = currentPage.iterator();
+                    } else {
+                        closeStream();
+                        return false;
+                    }
+
+                    pageIndex++;
+                } catch (IOException ex) {
+                    closeStream();
+                    logger.log(Level.WARNING, "A read error occured for file {0} page {1} {2}", new Object[]{cookieFile.getName(), pageIndex, ex}); //NON-NLS
+                    return false;
+                }
+            }
+
+            return currentIterator.hasNext();
+        }
+
+        /**
+         * Get the next cookie from current the current CookieIterator
+         *
+         * @return The next cookie
+         */
+        @Override
+        public Cookie next() {
+            // Just in case someone uses next without hasNext, this check will
+            // make sure there is more elements and that we iterate properly 
+            // through the pages.
+            if (!hasNext()) {
+                throw new NoSuchElementException();
+            }
+            return currentIterator.next();
+        }
+
+        /**
+         * Close the DataInputStream
+         */
+        private void closeStream() {
+            if (dataStream != null) {
+                try {
+                    dataStream.close();
+                    dataStream = null;
+                } catch (IOException ex) {
+                    logger.log(Level.WARNING, "SafariCookieReader unable to close DataInputStream for file {0} {1}", new Object[]{cookieFile.getName(), ex}); //NON-NLS
+                }
+            }
+        }
+    }
+
+    /**
+     * Wrapper class for an instance of a CookiePage in the binarycookie file
```

Comment:
We don't have a published standard for this, but I think it works best if we use periods to terminate sentences in our javadocs. The practice in this file is inconsistent.

The next time I have a little time on my hands, I will write javadocs guidelines for the team and community.